irohad/main/irohad.cpp

```diff
@@ -74,9 +88,19 @@ DEFINE_string(keypair_name, "", "Specify name of .pub and .priv files");
 DEFINE_validator(keypair_name, &validate_keypair_name);
 
 /**
- * Creating boolean flag for overwriting already existing block storage
+ * Creating enum flag for overwriting already existing block storage
+ */
+DEFINE_string(
+    blockstore_strategy,
+    "abort",
+    "behaviour in case of non-empty storage and passed genesis block, "
+    "'overwrite' to clear storage and apply new genesis block, "
+    "'continue' to continue with old storage and ignore new genesis block"
+    "or 'abort' to terminate with conflict error");
```

Comment:
Excuse me, but I still don't understand, what is the use case for 'abort' flag? Is that some kind of termination check, that we passed genesis block, when we did not need it, and it is obligatory to finish program execution?

Reply:
when genesis block is passed and flag is not set (default is abort). It terminates with an error to specify behaviour, just like it was before, when you run irohad second time on non-empty blockstore with passed genesis block